src/librustc/infer/region_inference/mod.rs

```diff
@@ -186,7 +187,7 @@ pub struct RegionVarBindings<'a, 'gcx: 'a+'tcx, 'tcx: 'a> {
     /// Constraints of the form `A <= B` introduced by the region
     /// checker.  Here at least one of `A` and `B` must be a region
     /// variable.
-    constraints: RefCell<FxHashMap<Constraint<'tcx>, SubregionOrigin<'tcx>>>,
+    constraints: RefCell<BTreeMap<Constraint<'tcx>, SubregionOrigin<'tcx>>>,
```

Comment:
Sorry, comment *here*

```
// Using `BTreeMap` because the order in which we iterate over these constraints
// can affect the way we build the region graph, which in turn affects the way
//  that region errors are reported, leading to small variations
// in error output across runs and platforms.
```

Reply:
https://github.com/rust-lang/rust/pull/44167/commits/bb60f30e3bada83634f462d47a5ec03413d1b458